src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/branch/BitbucketCloudBranch.java

```diff
@@ -25,29 +25,42 @@
 
 import com.cloudbees.jenkins.plugins.bitbucket.api.BitbucketBranch;
 import com.cloudbees.jenkins.plugins.bitbucket.client.repository.BitbucketCloudRepository;
+
+import java.util.List;
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 import org.codehaus.jackson.annotate.JsonCreator;
 import org.codehaus.jackson.annotate.JsonProperty;
 
 public class BitbucketCloudBranch implements BitbucketBranch {
     private final String name;
+    private final boolean isActive;
     private long dateInMillis;
     private String hash;
 
     @JsonCreator
-    public BitbucketCloudBranch(@Nonnull @JsonProperty("name") String name, @Nullable @JsonProperty("target") BitbucketCloudBranch.Target target) {
+    public BitbucketCloudBranch(@Nonnull @JsonProperty("name") String name,
+                                @Nullable @JsonProperty("target") BitbucketCloudBranch.Target target,
+                                @Nullable @JsonProperty("heads") List<Head> heads) {
         this.name = name;
         if(target != null) {
             this.dateInMillis = target.repo.getUpdatedOn() != null ? target.repo.getUpdatedOn().getTime() : 0;
             this.hash = target.hash;
         }
+
+        // For Hg repositories, Bitbucket returns all branches, including the closed/inactive ones.
+        // To determine if a branch has been closed, we look at the heads property:
+        // - Branches with non-empty heads are active.
+        // - Branches with empty heads are inactive.
+        // - On branches from git repositories heads is null. They all are active.
+        this.isActive = heads == null || !heads.isEmpty();
```

Comment:
@michaelneale 
On a git repo there are no heads, which `heads == null` covers while on hg repo a closed branch is empty while an open branch is not, so `!heads.isEmpty();` is correct for hg repo.

Check JSON for Git repo
https://api.bitbucket.org/2.0/repositories/caszum/testrepo/refs/branches
Check JSON for HG repo
https://api.bitbucket.org/2.0/repositories/caszum/testhgrepo/refs/branches

Reply:
thanks @casz reading more: https://stackoverflow.com/questions/8399501/how-to-delete-a-branch-in-bitbucket-with-mercurial-hg - close is very much delete. There is no delete in hg - you can strip commit, or probably some garbage collect, but not an explicit delete.